### PR TITLE
enhance: add resource group declarative api

### PR DESCRIPTION
**Pull request by chyezh**
issue: https://github.com/milvus-io/milvus/issues/32282

- Add `UpdateResourceGroups` and modify `AddResourceGroup` api.

- Add example for add resource group declarative api.

### CODE REVIEW

**Comment by chyezh** [comment]
Some code should not be merged into sdk.
reopen it after ready.

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        Map<String, ResourceGroupInfo.Builder> result = new HashMap<>();
+        for (String resourceGroupName : resourceGroups.getResourceGroupsList()) {
+            R<DescribeResourceGroupResponse> resourceGroupInfoResp = client.describeResourceGroup(
+                    DescribeResourceGroupParam.newBuilder().withGroupName(resourceGroupName).build());
+            DescribeResourceGroupResponse resourceGroupInfo = unwrap(resourceGroupInfoResp);
+
+            ResourceGroupInfo.Builder builder = ResourceGroupInfo.newBuilder()
+                    .withResourceGroupName(resourceGroupName)
+                    .withRequestsNodeNum(resourceGroupInfo.getResourceGroup().getConfig().getRequests().getNodeNum())
+                    .withConfig(new ResourceGroupConfig(resourceGroupInfo.getResourceGroup().getConfig()));
```
Comment (weiliu1031) [inline examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:71]: it's weird to pass rg config and requestNodeNum at same time

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            // skip if the collection is not loaded or is already added to resourceGroup.
+            if (currentResourceGroupName == null || currentResourceGroupName == resourceGroupName) {
+                continue;
+            }
+            R<RpcStatus> status = client.transferReplica(TransferReplicaParam.newBuilder()
+                    .withDatabaseName(dbName)
+                    .withCollectionName(collection)
+                    .withSourceGroupName(currentResourceGroupName)
+                    .withTargetGroupName(resourceGroupName)
+                    .withReplicaNumber(1L)
```
Comment (weiliu1031) [inline examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:218]: why just we transfer only one replica to target resource group?

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+        // Get the current resource group of the collection.
+        R<GetReplicasResponse> replicaResp = client
+                .getReplicas(GetReplicasParam.newBuilder().withCollectionName(collection).withDatabaseName(dbName).build());
+        GetReplicasResponse replicas = unwrap(replicaResp);
+
+        if (replicas.getReplicasCount() > 1) {
+            // Multi replica is now supported now.
+            throw new RuntimeException(String.format("Replica number {} is greater than 1, did not support now",
+                    replicas.getReplicasCount()));
```
Comment (weiliu1031) [inline examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:255]: multi replica is supported now

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        }
+        if (replicas.getReplicasCount() == 0) {
+            logger.warn("Collection {} @ Database {} has no replica, skip it", collection, dbName);
+            return null;
+        }
+
+        return replicas.getReplicasList().get(0).getResourceGroupName();
```
Comment (weiliu1031) [inline examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:262]: ditto

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java (hunk 1 above)
Comment (chyezh) [reply examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:71]: redundant code here, I will remove it.

File: examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java (hunk 2 above)
Comment (chyezh) [reply examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java:218]: It's an example. For easy coding, add a single replica limits here.
I will add a comment to explain it.

**Review by yelusion2 — APPROVED** [review]
/lgtm

**Pull request merged**